src/Jaeger.Transport.Thrift/Transport/JaegerUDPTransport.cs

```diff
@@ -1,24 +1,126 @@
+        /// <param name="sender">The ISender that will take care of the actual sending as well as storing the items to be sent</param>
+        /// <param name="serialization">The object that is used to serialize the spans into Thrift</param>
+        internal JaegerUdpTransport(int maxPacketSize, ISender sender, ISerialization serialization = null)
+            : base(sender, serialization)
+        {
+            // Each span is first written to thriftBuffer to determine its size in bytes.
+            _thriftBuffer = new TMemoryBuffer();
+            _thriftProtocol = _sender.ProtocolFactory.GetProtocol(_thriftBuffer);
+
+            if (maxPacketSize == 0)
+            {
+                maxPacketSize = TransportConstants.UdpPacketMaxLength;
+            }
+
+            _maxByteBufferSize = maxPacketSize - TransportConstants.UdpEmitBatchOverhead;
+        }
+
         /// <inheritdoc />
         /// <summary>
-        /// This constructor expects Jaeger running running on <value>DEFAULT_AGENT_UDP_HOST</value>
-        /// and port <value>DEFAULT_AGENT_UDP_COMPACT_PORT</value>
+        /// ProtocolAppendLogicAsync contains the specific logic for appending spans to be
+        /// sent over UDP. It will keep track of the byte size of the information to be sent
+        /// to the Jaeger Agent and make sure that it does not go over the max UDP packet size
+        /// by automatically flushing the buffer. It will throw and exception if the span is
+        /// too large to send in one UDP packet by itself.
         /// </summary>
-        public JaegerUdpTransport(int bufferSize = 0) : this(TransportConstants.DefaultAgentHost, TransportConstants.DefaultAgentUdpJaegerCompactThriftPort, bufferSize)
+        /// <param name="span">The span serialized in Jaeger Thrift</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        internal override async Task<int> ProtocolAppendLogicAsync(JaegerSpan span, CancellationToken cancellationToken)
         {
+            if (_processByteSize == 0) {
+                _processByteSize = await CalcSizeOfSerializedThrift(_process);
+                _byteBufferSize += _processByteSize;
+            }
+
+            var spanSize = await CalcSizeOfSerializedThrift(span);
+
+            if (spanSize > _maxByteBufferSize) {
+                throw new Exception("Span too large to send over UDP");
+            }
+
+            _byteBufferSize += spanSize;
+
+            // if the span fits in the buffer, buffer it
+            if (_byteBufferSize <= _maxByteBufferSize) {
+                _sender.BufferItem(span);
+
+                // if we can potentially fit more spans in the buffer don't flush
+                if (_byteBufferSize < _maxByteBufferSize) {
+                    return 0;
+                }
+
+                // can't fit anything else in the buffer, flush it
+                return await FlushAsync(cancellationToken);
```

Comment:
Looks like `_byteBufferSize` is not reset in this case. Maybe add `_byteBufferSize = _processByteSize`?

Reply:
Good catch. That woulda been a nice bug...